Signal/src/ViewControllers/GifPicker/GifPickerViewController.swift

```diff
@@ -295,36 +300,79 @@ class GifPickerViewController: OWSViewController, UISearchBarDelegate, UICollect
     // MARK: - UICollectionViewDelegate
 
     public func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+
         guard let cell = collectionView.cellForItem(at: indexPath) as? GifPickerCell else {
             owsFail("\(TAG) unexpected cell.")
             return
         }
-        guard let asset = cell.animatedAsset else {
-            Logger.info("\(TAG) unload cell selected.")
-            return
-        }
-        let filePath = asset.filePath
-        guard let dataSource = DataSourcePath.dataSource(withFilePath: filePath) else {
-            owsFail("\(TAG) couldn't load asset.")
+
+        guard cell.stillAsset != nil || cell.animatedAsset != nil else {
+            // we don't want to let the user blindly select a gray cell
+            Logger.debug("\(TAG) ignoring selection of cell with no preview")
             return
         }
-        let attachment = SignalAttachment(dataSource: dataSource, dataUTI: asset.rendition.utiType)
-        guard let thread = thread else {
-            owsFail("\(TAG) Missing thread.")
+
+        guard self.selectedCell == nil else {
+            owsFail("\(TAG) Already selected cell")
             return
         }
-        guard let messageSender = messageSender else {
-            owsFail("\(TAG) Missing messageSender.")
-            return
+        self.selectedCell = cell
+
+        // Fade out all cells except the selected one.
+        let maskingView = OWSBezierPathView()
```

Comment:
Is a `OWSBezierPathView` necessary? Could we just use a UIView and set it's background color and opacity?

Reply:
If we had a solid rectangular mask I would do that, but here we bunch a hole out of the mask, so that the **selected** view is not masked, but everything else is. 

Let me know if there's an easier way to do this.

Sorry I should have included a screenshot in the PR, here's what it looks like:

<img width="559" alt="screen shot 2017-10-20 at 9 52 51 am" src="https://user-images.githubusercontent.com/217057/31832604-75f035da-b57c-11e7-8e73-56f5275a70ff.png">